cmd/helm/repo_update.go

```diff
@@ -116,15 +116,22 @@ func (o *repoUpdateOptions) run(out io.Writer) error {
 
 func updateCharts(repos []*repo.ChartRepository, out io.Writer, failOnRepoUpdateFail bool) error {
 	fmt.Fprintln(out, "Hang tight while we grab the latest from your chart repositories...")
-	var wg sync.WaitGroup
-	var repoFailList []string
+
+	var (
+		wg           sync.WaitGroup
+		mu           sync.Mutex
+		repoFailList []string
+	)
+
 	for _, re := range repos {
 		wg.Add(1)
 		go func(re *repo.ChartRepository) {
 			defer wg.Done()
 			if _, err := re.DownloadIndexFile(); err != nil {
+				mu.Lock()
```

Comment:
We should use `defer mu.Unlock()` (if the above channel option is not used). Even while the code here is trivial, "manually" relying on `mu.Unlock()` being called is a footgun waiting to happen.

Reply:
fixed